build-support/bin/generate_travis_yml.py

```diff
@@ -485,6 +485,7 @@ def lint(python_version: PythonVersion) -> Dict:
         **linux_shard(python_version=python_version, install_travis_wait=True),
         "name": f"Self-checks and lint (Python {python_version.decimal})",
         "script": [
+            *_install_rust(),
```

Comment:
Nit, may want to put this in `pre_install`. I think you'd want to use our helper function `safe_extend()`.

Also, I wonder if we want to have the NATIVE_ENGINE_CACHE applied to cache the Rust install.

Reply:
The rust install here will run `cargo check` at the most... I don't think that the `NATIVE_ENGINE_CACHE` should be relevant to it.